### ISSUE TITLE
Add pytest-watch

### ISSUE THREAD

**Issue by de-code** [elifesciences/python-project-poc#3]
and a corresponding make command would be useful

**Comment by GiancarloFusiello** [comment]
Personally I'm not a fan of this. I don't want something running all of the time when I make changes but I can see how it could be useful.

I guess we could add a separate make command such as `make tests-watch` and `make t-watch` alias for this so it's an option for users that want this functionality.

@giorgiosironi @thewilkybarkid @gnott @lsh-0 thoughts?

**Comment by de-code** [comment]
Yes, I wouldn't want anyone to force using it. It's usually a separate command. It should also allow parameters to be passed through. e.g. I might only want to run tests within a package or use other selectors.

BTW running tests on every change is great. I think it encourages good practise, from keeping tests fast, making small incremental changes and helps writing tests first (write tests, see it fail, make it pass). IDEs may also have it built in, but `pytest-watch` seemed good enough so far.

**Comment by GiancarloFusiello** [comment]
@de-code what is the command/options you usually use with this? Just `ptw` or do you use anything else?

**Comment by de-code** [comment]
If the project and tests are small, then usually just `pytest-watch`. If I want to select tests, I might specify the package, e.g. `pytest-watch <directory>` or where I have a project which include slow tests, I might pass a filter to _pytest_, e.g. `pytest-watch -- -m "not slow"` (where such tests have been marked _slow_).

**Comment by GiancarloFusiello** [comment]
Thinking about this, I'm starting to think that this is outside the scope of this project. It's something that can be added to projects along with other commands that might be useful but the guidelines are for basic functionality, which is just to be able to run tests with a simple command. I think what has been added meets the criteria.

**Comment by de-code** [comment]
As part of the proof-of-concept (in particular poetry) it's certainly not required.

As a python project template, I would argue it's a basic functionality that I would want to use in any python project. Sure I could add it to the projects I am more involved in but it would make sense to just make it consistent.
